lib/lock.js

```diff
@@ -39,19 +40,9 @@ exports.reapLock = function(key, callback) {
  */
 exports.acquireLock = function(key, maximumLocks, ttlSeconds, callback) {
   var lockId = uuid.v1();
-  client.watch(key);
-  client.scard(key, function(err, cardinality) {
+  client.eval(lua.acquireLock, 1, key, lockId, maximumLocks, ttlSeconds, function(err, response) {
     if (err) { return callback(err, null); }
-    if (cardinality >= maximumLocks) { return callback(err, null); }
-    client.multi()
-      .sadd(key, lockId)
-      .setex(lockId, ttlSeconds, key)
-      .exec(function(err, replies) {
-        if (err) {
-          return callback(err, null);
-        }
-        return callback(err, replies ? lockId : null);
-      });
+    return callback(err, response ? lockId : null);
```

Comment:
Convention in this sort of scenario is to just return null as the first arg for clarity (to communicate you've handled the error).